contracts/src/TeleporterTokenDestination.sol

```diff
@@ -541,6 +346,215 @@ abstract contract TeleporterTokenDestination is
+        });
+
+        bytes32 messageID = _sendTeleporterMessage(
+            TeleporterMessageInput({
+                destinationBlockchainID: sourceBlockchainID,
+                destinationAddress: tokenSourceAddress,
+                feeInfo: TeleporterFeeInfo({
+                    feeTokenAddress: input.primaryFeeTokenAddress,
+                    amount: primaryFee
+                }),
+                requiredGasLimit: input.requiredGasLimit,
+                allowedRelayerAddresses: new address[](0),
+                message: abi.encode(message)
+            })
+        );
+
+        emit TokensSent(messageID, _msgSender(), input, amount);
+    }
+
+    /**
+     * @dev Sends tokens to the specified destination token bridge instance via multi-hop.
+     *
+     * Requirements:
+     *
+     * - {input.destinationBridgeAddress} cannot be the zero address
+     * - {input.recipient} cannot be the zero address
+     * - {amount} must be greater than 0
+     */
+    function _processSendMultiHop(SendTokensInput calldata input, uint256 amount) private {
+        _validateMultiHopInput(
+            input.destinationBlockchainID, input.destinationBridgeAddress, input.multiHopFallback
+        );
+
+        uint256 primaryFee;
+        (amount, primaryFee) = _prepareSend({
+            amount: amount,
+            primaryFeeTokenAddress: input.primaryFeeTokenAddress,
+            primaryFee: input.primaryFee,
+            secondaryFee: input.secondaryFee
+        });
+
+        BridgeMessage memory message = BridgeMessage({
+            messageType: BridgeMessageType.MULTI_HOP_SEND,
+            payload: abi.encode(
+                MultiHopSendMessage({
+                    destinationBlockchainID: input.destinationBlockchainID,
+                    destinationBridgeAddress: input.destinationBridgeAddress,
+                    recipient: input.recipient,
+                    amount: amount,
+                    secondaryFee: input.secondaryFee,
+                    secondaryGasLimit: input.requiredGasLimit,
+                    multiHopFallback: input.multiHopFallback
+                })
+                )
+        });
+
+        bytes32 messageID = _sendTeleporterMessage(
+            TeleporterMessageInput({
+                destinationBlockchainID: sourceBlockchainID,
+                destinationAddress: tokenSourceAddress,
+                feeInfo: TeleporterFeeInfo({
+                    feeTokenAddress: input.primaryFeeTokenAddress,
+                    amount: primaryFee
+                }),
+                requiredGasLimit: MULTI_HOP_REQUIRED_GAS,
+                allowedRelayerAddresses: new address[](0),
+                message: abi.encode(message)
+            })
+        );
+
+        emit TokensSent(messageID, _msgSender(), input, amount);
+    }
+
+    /**
+     * @dev Sends tokens to the specified recipient contract on the destination chain by
+     * calling the {receiveTokens} method of the respective recipient.
```

Comment:
If we decide to keep the call to `_validateSingleHopInput` inside this function, let's include the requirements in the doccomment.

Reply:
I thought we used the doc comment for requirements for validation that should be done before the function is called?

Reply:
Hmm, it seems like there's some inconsistencies in how we treate the requirements doccomment. For instance, the doccomment for `_send` in `TeleporterTokenDestination.sol` describes how the input is validated within the function call, not the assumptions that the function makes about its input. I'm cool deferring this in this PR, but we should take a pass through for comment consistency.